projects/mangol/src/lib/modules/map/map.component.ts

```diff
@@ -112,25 +113,31 @@ export class MapComponent implements OnInit, AfterViewInit, OnDestroy {
 
     // React to layer changes in the store
     this.layersSubscription = this.store
-      .select(state => state.layers.layers)
+      .select(fromMangol.getLayers)
       .subscribe((layers: MangolLayer[]) => {
         this.store
-          .select(state => state.map.map)
+          .select(fromMangol.getMap)
           .pipe(take(1))
           .subscribe(map => {
-            // Delete all previously loaded layers in the map
-            map.getLayers().forEach(l => {
-              map.removeLayer(l);
-            });
-            // Add all OL layers
-            layers.forEach(l => {
-              map.addLayer(l.layer);
-            });
+            // Add or remove OL layers depending on updates from store
+            const existingLayers = map.getLayers().getArray();
+            const tileLayers = layers.map(l => l.layer) as BaseLayer[];
+            tileLayers.length > existingLayers.length - 1 ?
+            tileLayers.forEach(l => {
+                if (existingLayers.indexOf(l) === -1) {
+                  map.addLayer(l);
+                }
+              }) :
+              existingLayers.forEach(l => {
+                if (l.getType() === 'TILE' && tileLayers.indexOf(l) === -1) {
```

Comment:
@fegyi001 
I don't think this is the best way to handle removing layers. Especially part with 'TILE', but I've been using only tile layers, that's why I have it this way. You can remove this change, but adding/removing layers dynamically is desired feature.